service/store/store.go

```diff
@@ -336,18 +336,18 @@ func getBid(reader ds.Read, id auction.BidID) (*Bid, error) {
 
 // SetAwaitingProposalCid updates bid with the given sources and switch status to BidStatusAwaitingProposal. If a bid is
 // not found for id, ErrBidNotFound is returned.
-func (s *Store) SetAwaitingProposalCid(id auction.BidID, sources auction.Sources) error {
+func (s *Store) SetAwaitingProposalCid(ctx context.Context, id auction.BidID, sources auction.Sources) error {
 	if err := sources.Validate(); err != nil {
 		return err
 	}
 
-	txn, err := s.store.NewTransaction(false)
+	txn, err := s.store.NewTransaction(ctx, false)
```

Comment:
Now `NewTransaction` in `go-datastore` receives a `ctx`. That's why it bubbles up from here to all our APIs to the "top of the stack".